src/main/java/io/cryostat/net/web/http/api/v2/TargetRecordingOptionsListGetHandler.java

```diff
@@ -108,6 +114,15 @@ public boolean isAsync() {
         return false;
     }
 
+    @Override
+    public SecurityContext securityContext(RequestParameters params) {
+        ConnectionDescriptor cd = getConnectionDescriptorFromParams(params);
+        return discoveryStorage
+                .lookupServiceByTargetId(cd.getTargetId())
```

Comment:
So if I am understanding this correctly, each endpoint is linked with a certain SecurityContext that each request needs to match or contain within a user-provided token in order for the request to be fulfilled, correct?

If I am getting this right, then why does the exception throw 404? If a SecurityContext object isn't found within Cryostat, isn't it the servers fault and not the clients?

Reply:
Each API endpoint is able to determine the correct Security Context for a given request. The AuthManager checks if the user token associated with the request has the correct permissions for the specific request security context. The user token is the usual one passed by the `Authorization` header.

This throws a 404 because actually there is always a Security Context for a request, since they are generated by looking at the resources (targets, recordings, etc.) affected by the request. So if this chunk of logic falls down to that `orElseThrow` case then the cause is actually that the `lookupServiceByTargetId` call returned an empty Optional, ie the request was for an unknown target, therefore it's a 404.